### PR TITLE
Remove the requirement that DependencyCollector getters are named after Configurations

### PR DESCRIPTION
**Pull request by tresat**
This is more correct as it is not required to name them after Configurations.  Instead, check if the type they are implemented on extends Dependencies.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/dependencyConfigurationSchema.kt [hunk 1]
```diff
@@ -101,9 +102,8 @@ class DependencyFunctionsExtractor(val configurations: DependencyConfigurations)
 private
 class ImplicitDependencyCollectorFunctionExtractor(val configurations: DependencyConfigurations) : FunctionExtractor {
```
Comment (jvandort) [inline platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/dependencyConfigurationSchema.kt:103]: This method no longer needs the configuration container as an argument

File: platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/dependencyConfigurationSchema.kt [hunk 2]
```diff
@@ -185,14 +185,17 @@ class ImplicitDependencyCollectorFunctionResolver(configurations: DependencyConf
 
     private
     fun getDependencyCollectorGetter(receiverClass: KClass<*>, configurationName: String): KFunction<*>? = receiverClass.functions
```
Comment (jvandort) [inline platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/dependencyConfigurationSchema.kt:187]: This method still operates on configuration names. Does this also need to be updated?

**Comment by tresat** [comment]
@bot-gradle test this

**Review by h0tk3y — APPROVED** [review]
LGTM 👍

An idea of a next step: if the dependency DSL types must extend `Dependencies`, we might want to use their `ProjectDependency project(String projectPath)` as the value factory to create project dependencies instead of using our own [hacky fake function](https://github.com/gradle/gradle/blob/c8ed6db4810f526ce774d895a45f136c012b694a/platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/ProjectTopLevelReceiver.kt#L34). It's too early to remove the latter, but we can introduce the former.

**Comment by tresat** [comment]
> LGTM 👍
> 
> An idea of a next step: if the dependency DSL types must extend `Dependencies`, we might want to use their `ProjectDependency project(String projectPath)` as the value factory to create project dependencies instead of using our own [hacky fake function](https://github.com/gradle/gradle/blob/c8ed6db4810f526ce774d895a45f136c012b694a/platforms/core-configuration/declarative-dsl-provider/src/main/kotlin/org/gradle/internal/declarativedsl/project/ProjectTopLevelReceiver.kt#L34). It's too early to remove the latter, but we can introduce the former.

Good idea.  I hope to get to this, and Property support (in additional to function getters) for `DependencyCollector`, soon.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**